src/org/infoglue/calendar/controllers/EventController.java

```diff
@@ -1993,24 +1994,58 @@ public boolean hasUserGroup(InfoGluePrincipalBean principal, Event event) throws
         
         return false;
     }
-    
-    public List getAssetKeys()
-    {
-        List assetKeys = new ArrayList();
-        
-        int i = 0;
-        String assetKey = PropertyHelper.getProperty("assetKey." + i);
-        while(assetKey != null && assetKey.length() > 0)
-        {
-            assetKeys.add(assetKey);
-            
-            i++;
-            assetKey = PropertyHelper.getProperty("assetKey." + i);
-        }
-        
-        return assetKeys;
-    }
 
+	/**
+	 * Calls {@linkplain #getAssetKeys(Event)} with null as Event.
+	 * @return
+	 */
+	public List getAssetKeys()
+	{
+		return getAssetKeys(null);
+	}
+
+	private boolean isUndefinedAssetKey(String assetKey)
+	{
+		return assetKey.startsWith("@assetKey") && assetKey.endsWith("@");
+	}
+
```

Comment:
This method could also call the new ResourceController.isUndefinedProperty() method since it checks for null and the empty string too.